src/Cli/dotnet/OptionForwardingExtensions.cs

```diff
@@ -17,8 +17,8 @@ public static class OptionForwardingExtensions
 
         public static Option ForwardAsSingle<T>(this Option<T> option, Func<T, string> format) => new ForwardedOption<T>(option).SetForwardingFunction(format);
 
-        public static Option ForwardAsProperty(this Option<string> option) => new ForwardedOption<string>(option)
-            .SetForwardingFunction((o) => new string[] { $"{option.Aliases.FirstOrDefault()}:{o.Replace("roperty:", string.Empty)}" });
+        public static Option ForwardAsProperty(this Option<string[]> option) => new ForwardedOption<string[]>(option)
+            .SetForwardingFunction((optionVals) => optionVals.SelectMany(optionVal => new string[] { $"{option.Aliases.FirstOrDefault()}:{optionVal.Replace("roperty:", string.Empty)}" }));
```

Comment:
What would happen if you passed something like `/p:WeirdProperty=property:value`?  Wouldn't this parsing mess up the value?

Reply:
Yes, it would. The string replace is a work around for a system command line behavior (bug?) where the first option alias is matched even if a better match exists. So with the property option, we have the aliases `-p` and `-property` but since `-p` is the prefix in both cases it's always the alias that's matched. For example, with the input `-property:prop=true` the token result is option: `-p` and option value: `roperty:prop=true`

cc @jonsequitur

Reply:
Will track this issue with https://github.com/dotnet/sdk/issues/14461, but in the meantime I don't think it effects this change.

Reply:
The System.CommandLine behavior is by design. 

> An option and its argument may or may not appear as separate tokens. (In other words, the whitespace separating them is optional.) Thus, ‘-o foo’ and ‘-ofoo’ are equivalent.

https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html

Reply:
I think MSBuild doesn't follow the GNU spec.  It doesn't require two dashes before the long form of an argument (`-property`), and it does require a separator (colon) between the option and the argument.  Does System.CommandLine intend to support that behavior to some degree so we can maintain compatibility with arguments that are passed through to MSBuild?